### PR TITLE
[Delivers #93637304] improve email grouping

### PR DESCRIPTION
**Pull request by afeld**
The goal is to get all emails related to a request to show up in a single thread in GMail and other mail clients.

### CODE REVIEW

**Comment by afeld** [comment]
:zap: 

screen shot 2015-06-10 at 1 47 04 am

File: app/decorators/proposal_decorator.rb [hunk 1]
```diff
@@ -55,4 +55,12 @@ def completed_status_message
   def progress_status_message
     "#{number_approved} of #{total_approvers} approved."
   end
+
+  def email_subject
+    "Request #{proposal.public_identifier}"
```
Comment (afeld) [inline app/decorators/proposal_decorator.rb]: Open to suggestions about this. The format suggested in [the story](https://www.pivotaltracker.com/story/show/93637304) is more likely to change during the lifetime of a request and thus more likely to break the threading...not sure what the right balance is. Some ideas, from most to least volatile/safe:

1. `Request #<ID>`
1. `Request #<ID> – <Proposal name>`
1. `Request #<ID>, <Org Code>, <Building code>, from <Requester>`

File: spec/mailers/communicart_mailer_spec.rb [hunk 2]
```diff
@@ -209,16 +215,14 @@ def sender_names(mail)
...
+    it_behaves_like "a Proposal email"
 
     it 'renders the receiver email' do
       expect(mail.to).to eq([proposal.requester.email_address])
     end
 
-    it 'renders the sender email' do
-      expect(mail.from).to eq(["reply@stub.gov"])
+    it "sets the sender name" do
+      expect(sender_names(mail)).to eq([nil])
```
Comment (cmc333333) [inline spec/mailers/communicart_mailer_spec.rb]: I'm not sure what this and the `it "sets the sender name"` above are actually testing. Seems like `it_behaves_like "a Proposal email"` covers the sender?

**Comment by cmc333333** [comment]
Big improvement! Only one small suggestion

**Comment by afeld** [comment]
Updated!

**Pull request merged**